parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/ColumnIndexBuilder.java

```diff
@@ -368,6 +369,20 @@ public <T extends Comparable<T>> PrimitiveIterator.OfInt visit(NotIn<T> notIn) {
       return IndexIterator.all(getPageCount());
     }
 
+    @Override
+    public <T extends Comparable<T>> PrimitiveIterator.OfInt visit(Contains<T> contains) {
```

Comment:
I need to think about the behavior here one -- `visit(And and)` and `visit(Or or) ` in `ColumnIndexBuilder` throw an UnsupportedOperationException, so I just copied that behavior for composed Contains predicates. But I'm thinking we can actually support this by taking the union/intersection of the `PrimitiveIterator.OfInt`, kind of similar to how `ColumnIndexFilter` [works](https://github.com/apache/parquet-mr/blob/apache-parquet-1.14.0-rc0/parquet-column/src/main/java/org/apache/parquet/internal/filter2/columnindex/ColumnIndexFilter.java#L193-L209).

Reply:
CMIW, throwing exception here might prohibit `contains` predicate to be used to filter parquet files with page index available.

Reply:
Makes sense. I'll look into doing an intersection/union operation here. Or otherwise just fall back to not filtering and returning `IndexIterator.all`.

Reply:
Updated the PR with implementations for each. I implemented them as new `PrimitiveIterator.OfInt`s that zip together the component left & right hand sides, deduplicating/intersecting/merging as needed. Implementing custom iterators is kind of error prone though... if this adds too much complexity I can always save it for a followup PR, or just materialize the iterators into `List`s which are a lot easier to work with 😅